src/bit-systems/quack.ts

```diff
@@ -0,0 +1,20 @@
+import { HubsWorld } from "../app";
+import { Held, Quack } from "../bit-components";
+import { defineQuery, enterQuery } from "bitecs";
+import { SOUND_QUACK, SOUND_SPECIAL_QUACK } from "../systems/sound-effects-system";
+
+const heldQuackQuery = defineQuery([Quack, Held]);
+const heldQuackEnterQuery = enterQuery(heldQuackQuery);
+
+const quack = (eid: number) => {
```

Comment:
Not ideal to have a whole system for managing this component but necessary for now as far as I can tell.

Reply:
Yeah I don't think this fits well anywhere else so I think it's ok for now. I'd consider inlining this function in the main system loop to avoid a function call.